internal/storage/user.go

```diff
@@ -0,0 +1,34 @@
+package storage
+
+import (
+	"context"
+	"github.com/jsfan/hello-neighbour/internal/config"
+	"github.com/jsfan/hello-neighbour/internal/storage/dal"
+	"github.com/jsfan/hello-neighbour/internal/storage/models"
+)
+
+func setupContextAndDbConnection(ctx context.Context) (ctext context.Context, cancelCtx context.CancelFunc, db *DBConnection) {
+	ctext, cancelCtx = context.WithCancel(ctx)
+	db = ctx.Value(config.DatabaseConnection).(*DBConnection)
+	return ctext, cancelCtx, db
+}
+
+func (conn *DBConnection) GetUserByEmail(ctx context.Context, email string) (user *models.UserProfile, errVal error) {
+	ctx, cancelCtx, db := setupContextAndDbConnection(ctx)
+	dbAccess, commitFunc, err := dal.GetDal(ctx, db.Db)
+	defer func() {
+		if err := commitFunc(); err != nil && errVal == nil {
+			errVal = err
+		}
+	}()
```

Comment:
Why do we use `commitFunc` here if we are just reading from the db?

Reply:
The way I have set it up right now, it uses a transaction every time. I don't think there is harm in that even if for pure reads it wouldn't be necessary.